microceph/ceph/config.go

```diff
@@ -3,15 +3,144 @@ package ceph
 import (
 	"context"
 	"database/sql"
+	"encoding/json"
 	"fmt"
 	"os"
 	"path/filepath"
 	"strings"
 
+	"github.com/canonical/microceph/microceph/api/types"
 	"github.com/canonical/microceph/microceph/common"
 	"github.com/canonical/microceph/microceph/database"
 )
 
+type ConfigExtras struct{
+	Who     string // Ceph Config internal <who> against each key
+	Daemons []string // Daemons that need to be restarted post config change.
+	Regexp  string // Regular Expression to check value against key
+	Cb      *func(s common.StateInterface, c types.Config)
+}
+
+type ConfigItem map[string]ConfigExtras
+
+// Check if certain key is present in the map.
+func (c ConfigItem) isKeyPresent(key string) bool {
+	if _, ok := c[key]; !ok {
+        return false
+    }
+
+	return true
+}
+
+// Return keys of the given set
+func (c ConfigItem) Keys() (keys []string) {
+    for k := range c {
+        keys = append(keys, k)
+    }
+    return keys
+}
+
+func GetConfigTable() ConfigItem {
+	return ConfigItem{
+		"public_network":  {"global", []string{"mon", "osd"}, "", nil},
+		"cluster_network": {"global", []string{"osd"}, "", nil},
+	}
+}
+// Instantiation of config table.
+var configTable = GetConfigTable()
+
+// Struct to get Config Items from config dump json output.
+type ConfigDump []struct{
+	Section string
+	Name    string
+	Value   string
+}
+
+func RestartCephDaemon(daemon string) error {
+	return snapReload(daemon)
```

Comment:
As is, I don't think that just wrapping `snapReload` is that useful; however, this might be a good place to restart and then wait for the daemon to be fully functional again

Reply:
Ack, snapReload has been changed to snapRestart with additional input param to do reloads, since just reloading the osd daemons was not updating the conf.